### PR TITLE
Fix: text chunking processor ingestion bug on multi-node cluster

### PR DESCRIPTION
**Pull request by yuye-aws**
### Description
For multi node cluster, the text chunking processor would produce "no such index" error if the configured shard number is less than the number of nodes. This is because some node does not contain the shard information. When we get max token count setting, `indicesService` fails to find the index information.
```
IndexService indexService = indicesService.indexServiceSafe(indexMetadata.getIndex());
```

### Issues Resolved
Fix ingestion bug on multi-node cluster

### Check List
- [ ] New functionality includes testing.
    - [ ] All tests pass
- [ ] New functionality has been documented.
    - [ ] New functionality has javadoc added
- [x] Commits are signed as per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by yuye-aws** [comment]
Hi maintainers. This PR is a fix towards text chunking processor. Please attach **backport 2.x** and **backport 2.13** labels to this PR.

**Comment by yuye-aws** [comment]
This PR is still work in progress. Before getting merged, this PR must satisfy two conditions :
1. All integration test cases get passed with a three node cluster.
2. All BWC tests for text chunking processor with a single shard number can still get passed.

**Comment by yuye-aws** [comment]
@model-collapse @zane-neo This PR is ready for review now. Please merge this PR after passing all the CI workflow.

**Review by vibrantvarun — DISMISSED** [review]

**Comment by chishui** [comment]
Shall we add an IT to cover this "configured shard number is less than the number of nodes" scenario? Can be done in a separate issue and PR.

**Comment by zhichao-aws** [comment]
> Shall we add an IT to cover this "configured shard number is less than the number of nodes" scenario? Can be done in a separate issue and PR.

In current CI all IT are run with one node. I think we can enhance the CI framework by adding the build with `-PnumNodes=3`. This can help us exclude bugs in distributed scenerio at early stage.

**Comment by yuye-aws** [comment]
> I think we can enhance the CI framework by adding the build with `-PnumNodes=3`

Good point. We can follow the same process like ml-commons.

File: src/main/java/org/opensearch/neuralsearch/processor/TextChunkingProcessor.java
Comment (zane-neo) [inline src/main/java/org/opensearch/neuralsearch/processor/TextChunkingProcessor.java:156]:
```suggestion
        return IndexSettings.MAX_TOKEN_COUNT_SETTING.getDefault(indexMetadata.getSettings());
```

Comment (yuye-aws) [reply src/main/java/org/opensearch/neuralsearch/processor/TextChunkingProcessor.java]: Already updated

Comment (yuye-aws) [reply src/main/java/org/opensearch/neuralsearch/processor/TextChunkingProcessor.java]: It takes me a while to test this recommended change.

Comment (yuye-aws) [reply src/main/java/org/opensearch/neuralsearch/processor/TextChunkingProcessor.java]: Unfortunately, this code change always return the default value for max token count.

**Comment by yuye-aws** [comment]
@zane-neo The current gradle checks get failed due to model deployed issue. Is it attributed to the latest update in  **ml-commons**, like async http client?

File: src/main/java/org/opensearch/neuralsearch/processor/TextChunkingProcessor.java
Comment (zane-neo) [reply src/main/java/org/opensearch/neuralsearch/processor/TextChunkingProcessor.java:156]: Can we check if there's any other method like `getDefault` we can use here? I don't think using explicit string `index.analyze.max_token_count` is a good practice, better to use the enum `IndexSettings.MAX_TOKEN_COUNT_SETTING`.

Comment (yuye-aws) [reply src/main/java/org/opensearch/neuralsearch/processor/TextChunkingProcessor.java]: Already updated

**Review by zane-neo — APPROVED** [review]

**Review by model-collapse — APPROVED** [review]
All concerns addressed.

**Comment by vibrantvarun** [comment]
Even gradle checks are failing @yuye-aws

**Comment by vibrantvarun** [comment]
"Model not deployed yet" error coming from ml-commons [https://github.com/opensearch-project/ml-commons/issues/2382](https://github.com/opensearch-project/ml-commons/issues/2382)

**Comment by zane-neo** [comment]
> "Model not deployed yet" error coming from ml-commons [opensearch-project/ml-commons#2382](https://github.com/opensearch-project/ml-commons/issues/2382)

This is another issue that related to ml-commons main branch, we'll track this with a new issue.

**Pull request merged**
